### PR TITLE
fix(orchestrator): store diff_head_sha unconditionally for commit pinning

### PR DESCRIPTION
**Pull request by RC918**
# fix(orchestrator): store diff_head_sha unconditionally for commit pinning

## Summary

Fixes `commit_id: null` issue in GitHub review posting that was causing `commit_pinning_attempted=False` in diagnostic logs.

**Root cause**: `diff_head_sha` was only stored inside the `if diff_content:` block in `reviewer_node`. When `diff_content` was empty (e.g., lockfile-only PRs, truncation edge cases), `diff_head_sha` was never stored in state, causing `publisher_node` to pass `commit_id=None` to `post_pr_review()`.

**Fix**: Move `diff_head_sha` storage outside the `if diff_content:` block so it's stored unconditionally whenever available. This ensures commit pinning works regardless of diff content state.

**Changes**:
- `reviewer_node`: Store `diff_head_sha` unconditionally (moved before `if diff_content:` block)
- `reviewer_node`: Add diagnostic log when storing `diff_head_sha`
- `reviewer_node`: Add warning log when `diff_head_sha` is null (commit pinning disabled)
- `publisher_node`: Add diagnostic log to verify `stored_head_sha` presence
- `publisher_node`: Emit both `sanitized_diff_length` and `diff_content_length` (legacy) for backward compatibility
- `test_graph.py`: Fix 6 tests that were failing due to PR #3086's `value_gate_dry_run` default change

## Updates since last revision

1. Added `TRUST` comment confirming `diff_head_sha` originates from GitHub API (`pr.head.sha` via `get_pr_diff`)
2. Renamed `diff_content_length` to `sanitized_diff_length` for clarity, with backward-compatible legacy key
3. Added warning log when `diff_head_sha` is null to improve debuggability
4. **Fixed CI test failures**: Added `@patch.dict(os.environ, {"VALUE_GATE_DRY_RUN": "true"})` to 6 tests that need to bypass Value Gate to test Rate Limiter logic (per CTO decision: tests should explicitly disable Value Gate when testing Rate Limiter)

## Review & Testing Checklist for Human

- [ ] **Verify logic change is correct**: The `diff_head_sha` storage was moved from inside `if diff_content:` to outside it. Confirm this doesn't break any other code paths that depend on the previous behavior.
- [ ] **Test with actual PR**: Deploy to staging and trigger a review on a PR. Check logs for:
  - `[Reviewer] Stored diff_head_sha for commit pinning` with non-null `diff_head_sha`
  - `[Publisher] Retrieved state for commit pinning` with non-null `stored_head_sha`
  - `[GitHub] DIAGNOSTIC: Final payload` with non-null `commit_id`
- [ ] **Test edge case**: If possible, test with a lockfile-only PR to verify `diff_head_sha` is stored even when `diff_content` is empty
- [ ] **Verify test fix is appropriate**: The 6 fixed tests now explicitly disable Value Gate via env var. Confirm this is the correct isolation approach for testing Rate Limiter logic.

### Notes

- All 42 CI checks now pass (test failures from PR #3086's `value_gate_dry_run` default change have been fixed in this PR)
- The C901 complexity warnings in lint are pre-existing (not introduced by this PR)
- This fix was identified through v1.2.0 diagnostic logging which showed `commit_id: null` and `commit_pinning_attempted=False`

### Follow-up Issues Created

- #3093: Add automated tests for empty diff_content edge case
- #3094: Document commit pinning flow and diff_head_sha dataflow
- #3095: Investigate CI/local test parity for github module

---
Link to Devin run: https://app.devin.ai/sessions/704acc3d97124d00a7e98394ff2203e1
Requested by: Ryan Chen (@RC918)

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/langgraph_orchestrator.py [hunk 1]
```diff
@@ -3747,6 +3747,20 @@ def reviewer_node(state: AgentState) -> AgentState:
...
                         )
 
+                    # Phase 2: Store head_sha for commit pinning (UNCONDITIONALLY)
+                    # This is critical for preventing 422 errors from race conditions
+                    # where new commits are pushed between diff generation and review posting.
+                    # Previously this was inside `if diff_content:` which caused commit_id: null
+                    # when diff_content was empty (e.g., lockfile-only PRs, truncation edge cases).
+                    # Fix: Store head_sha regardless of diff_content to ensure commit pinning works.
+                    if diff_head_sha:
+                        state["diff_head_sha"] = diff_head_sha
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/langgraph_orchestrator.py]: IMPACT: While functional, this pattern assumes `diff_head_sha` is trustworthy without validation. If `diff_head_sha` ever comes from user input or an untrusted source, it could lead to security or integrity issues. SUGGESTED FIX: Consider adding a brief comment confirming `diff_head_sha` originates from a trusted upstream source (e.g., GitHub API) to reinforce code safety.

File: handoff/20250928/40_App/orchestrator/langgraph_orchestrator.py [hunk 2]
```diff
@@ -4228,11 +4238,22 @@ def publisher_node(state: AgentState) -> AgentState:
...
 
+    # DIAGNOSTIC: Log stored_head_sha presence for commit pinning verification
+    logger.info("[Publisher] Retrieved state for commit pinning", extra={
+        "operation": "publisher",
+        "trace_id": trace_id,
+        "pr_number": pr_number,
+        "stored_head_sha": stored_head_sha[:8] if stored_head_sha else None,
+        "has_diff_content": bool(diff_content),
+        "diff_content_length": len(diff_content) if diff_content else 0
+    })
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/langgraph_orchestrator.py]: IMPACT: The log field `diff_content_length` may be misleading because it logs the length of already-truncated or sanitized content, not the original diff size. This could confuse debugging of truncation-related issues. SUGGESTED FIX: Rename to `sanitized_diff_length` or add `original_diff_length` if available to clarify what is being measured.

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review
